### ISSUE TITLE
Unsynced folders are lost, if you uncheck

### ISSUE THREAD

**Issue by aronpapp** [nextcloud/desktop#1238]
### Expected behaviour
* Warn user of possible data loss
* Ask in dialog box, if the user really wants to erase data or inform user, that unsynced folders were not removed from the filesystem

### Actual behaviour
Erases (unsynced) data silently.

### Steps to reproduce
1. Move or copy a folder to your sync folder using a filemamanger
2. Open nextCloud client
3. Do not wait to sync the new folder to the server
4. Uncheck checkbox for the new folder
5. Click Apply

### Client configuration
Client version: Version 2.5.1final (build 20181204)

Operating system: Debian 9.8

OS language: en

Qt version used by client package (Linux only, see also Settings dialog):
I guess it uses bundled Qt.

Client package (From Nextcloud or distro) (Linux only):
Nextcloud-2.5.1-x86_64.AppImage

Installation path of client:
No need to install

### Server configuration

Issue is not related to the server.

**Comment by Germano0** [comment]
What does it mean `Remove checkbox before new folder` ?

**Comment by aronpapp** [comment]
> What does it mean `4. Remove checkbox before new folder` ?

Sorry, I mean uncheck the cheeckbox (also fixed in the initial post)

**Comment by camilasan** [comment]
You mean you uncheck the folder so it doesn't get synced, correct?

**Comment by aronpapp** [comment]
> You mean you uncheck the folder so it doesn't get synced, correct?

Yes, it does not get synced, and gets deleted from the filesystem, so data is completly flushed.

**Comment by freechelmi** [comment]
That sounds like a very important problem.

**Comment by b12f** [comment]
There's a large discussion about this behaviour in the owncloud repo: https://github.com/owncloud/client/issues/2502

Includes more failure cases and a couple of proposed UI fixes.

**Comment by salagula** [comment]
That is really critical - we have exactly that problem. It dramatically lowers down the tool acceptance and. As the desktop app is available as source code: has someone tried to disable the method that deletes the local files? Any indication on the source code file where we can turn it off?

**Comment by er-vin** [comment]
I guess there's something I don't get regarding that one, but here unchecking something leads to this warning:
![Screenshot_20201022_125320](https://user-images.githubusercontent.com/1411800/96862446-c4be7280-1465-11eb-92f8-f17b230db65d.png)

And it only removes if you confirm it. To me, it's pretty clear it's going to kill data in there.

**Comment by b12f** [comment]
The problem is that it also removes files that aren't synced yet. So let's say you added a big directory to nextcloud, but want to not sync it (for now, or not after all), it removes the data completely. When would this happen? Mostly during installation; you select a folder in which the data is that you want to sync, but then find out there's a couple of large files in there that you do not want to sync after all. Poof, data gone.

From a UX perspective, the warning is not enough imo. This behaviour can lead to unsynchronized data being lost forever, so this should be a big fat red banner if there are unsynchronized files to be deleted. Even better, it should be possible to unsynchronize without deletion.

**Comment by salagula** [comment]
I agree to b12f - it is not about the message, it is about the fact that it deletes files before they are synced. Again my question: can someone point me to the code segment where this is triggered? I would like to comment the section out (if possible)

**Comment by er-vin** [comment]
> I agree to b12f - it is not about the message, it is about the fact that it deletes files before they are synced. Again my question: can someone point me to the code segment where this is triggered? I would like to comment the section out (if possible)

It's not that simple, it's really just about putting those folders in the selective sync exclude list and then the sync engine "doing its thing". It's not like there's a function where those folders are removed, this is deeply buried in the sync engine behavior AFAICT.

**Comment by irisv** [comment]
> I guess there's something I don't get regarding that one, but here unchecking something leads to this warning:
> ![Screenshot_20201022_125320](https://user-images.githubusercontent.com/1411800/96862446-c4be7280-1465-11eb-92f8-f17b230db65d.png)
> 
> And it only removes if you confirm it. To me, it's pretty clear it's going to kill data in there.

I'd like to vote for a dialogue that rather than cancel / apply gives three options:

1. remove folder from sync and delete local data (= the current behaviour)
2. remove folder from sync but keep local data (= my preferred behaviour in most cases)
3. cancel

**Comment by andreaslink-de** [comment]
> 1. remove folder from sync and **delete** local data (= the current behaviour)
> 2. remove folder from sync but **keep** local data (= my preferred behaviour in most cases)
> 3. cancel

Honestly, there is a big risk of doing no. 2 based on my own experience. I had cases where I removed a folder like this (might have be in older owncloud times) and I kept the folder in there but it was not synced anymore. After around x days I forgot about it and changed some important data on one machine in that folder and I changed files on the other machine some days later as well. It took me some further days to figure out, I was working on different instances and need to do manual sync work now. 

So the rule should be, if it is in your locally synced nextcloud folder, then it's synced and if it is not synced it is not in there - store your data somewhere else, but don't confuse yourself with folders being in sync and some not.

So no. 1 and no. 3 are the only realistic options for not messing up with your personal data controlling, so I disagree and would be surprised you describe a usecase where no. 2 is your "most preferred behaviour".

Where I agree is that there could be an additional warning, if data is going to be deleted, which was _not synced_ yet. 
Use case could be you move a bigger folder from anywhere to your local nextcloud folder to get it synced and expect the sync to be finished after a while, but due to low internet speed the folder was only half synced yet. If you then just try to remove the hook to get the folder deleted (maybe to save some local space) you have to be warned. There it makes sense for not loosing unsynced data.

**Comment by sojer** [comment]
Will a fix be included in the next nextcloud release?
Or is there something planned for a future release?

**Comment by irisv** [comment]
> So no. 1 and no. 3 are the only realistic options for not messing up with your personal data controlling, so I disagree and would be surprised you describe a usecase where no. 2 is your "most preferred behaviour".

Ok, then let me explain my usage to you. I have a set of core folders that need to be synced and kept up to date continuously. Those will never be deleted from the client, so those don't come into this discussion.
When I delete a folder form the client, it is -by definition- a folder I've only had use for temporary and now no longer need. So your scenario where _"around x days [you] forgot about it and changed some important data on one machine (...)"_  will never apply to these folders, because these are not files that are going to be changed. These are folders that I only needed to use for a limited period of time, that might have been handy to sync while they're useful but don't need to be synced any more once I'm done with whatever task they were created for. 
Still, sometimes I want to hang on to them locally, if only for a couple of minutes, because, well, deleting a whole folder in one go can be scary. What I do now is rename/move the folder, then remove it from the client and then figure out myself what want to delete and what needs to be archived & where. 

But from this discussion (and [the post that lead me here today](https://help.nextcloud.com/t/unsynced-folders-are-lost-if-you-uncheck-issue-1238/99285/1)) I conclude there are people out there who are not taking that extra precaution and are then surprised that files have 'disappeared'. Disappearing files - even if in reality its merely the _impression to some users_ that files have disappeared (even though they should have know that their actions would have this result), is just about the worst thing that can happen with a data storage solution. So something to be avoided at all cost, I'd think. 

So more importantly than my specific way of using the client of my data management routines in general, is that _forcing users to choose_ between deleting the local files or keeping them, is a way of hammering home the point that local files will be deleted. And if they're not yet on the server, that means they're gone for good. That's basically why I thought it'd be a good idea to introduce that additional option. Sometimes its useful to force people to make a choice. Yes, people should know that already, but when you have people asking whether this will be 'fixed' in future releases, you can conclude that this isn't behaviour that is intuitively understood by all users. So my suggestion was a way of giving those people that extra nudge they need.

**Comment by dirdi** [comment]
> Will a fix be included in the next nextcloud release?
> Or is there something planned for a future release?

Since there has not been reached a consensus about _how_ the current behavior should be altered, this is very unlikely.

---

That said, I would like to suggest an alternative to the proposals that have been made so far:
**Add a third option to move the folder out of the sync folder**.

If the users choose this option, the client would fire up a "choose directory" dialog and then move the not-yet-synced-folder out of the synced folder to the folder chosen by the client.

**Comment by irisv** [comment]
> That said, I would like to suggest an alternative to the proposals that have been made so far:
> **Add a third option to move the folder out of the sync folder**.
> 
> If the users choose this option, the client would fire up a "choose directory" dialog and then move the not-yet-synced-folder out of the synced folder to the folder chosen by the client.

That'd also work for me. That's basically what I do now manually.

**Comment by robballan** [comment]
I’d add a 4th option that seems more intuitive to me: sync anything put in a Nextcloud folder BEFORE deleting the local copy that is unchecked for syncing. That way the cloud copy exists but the local copy does not. 

For those who wish to keep a static local copy of some or all of the files — they shouldn’t need in Nextcloud folder to start with.

**Comment by sadr0b0t** [comment]
Maybe it should just forbid unchecking folders that are not yet synced to server with some kind of explanation why this checkbox is grayed out in disabled state and suggestion to move this folder out of sync folder manually in file manager.

The warning about removing synced folder from local file system is NOT the same as removing non-synced folder from local file system. In the 1-st case deleting data is safe. And in the 2-nd case deleting data is NOT safe. I think that removing local unsynced data is not a valid option, application should not do it in any case and should not suggest it. If user really wants to lose his data, he can just remove it in file manager manually.

**Comment by robballan** [comment]
You should NOT be able to uncheck a folder for syncing UNTIL it has been synced upstream. That would solve it.

> On Nov 25, 2020, at 9:13 AM, Anton <notifications@github.com> wrote:
> 
> 
> Maybe it should just forbid unchecking folders that are not yet synced to server with some kind of explanation why this checkbox is grayed out in disabled state and suggestion to move this folder out of sync folder manually in file manager.
> 
> The warning about removing synced folder from local file system is NOT the same as removing non-synced folder from local file system. In the 1-st case deleting data is safe. And in the 2-nd case deleting data is NOT safe. I think that removing local unsynced data is not a valid option, application should not do it in any case and should not suggest it. If user really wants to lose his data, he can just remove it in file manager manually.
> 
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub, or unsubscribe.
> 

_____

raa

**Comment by wsw70** [comment]
> You should NOT be able to uncheck a folder for syncing UNTIL it has been synced upstream. That would solve it.

@b12f gives a very good case where this would not work: the initial setup of Nextcloud. The sync starts, you realize that there is a 200 GB subfolder you do not want to synchronize, uncheck it and the data is gone.

Not removing unsynced data is a solution that is safer.

**Comment by sadr0b0t** [comment]
> The sync starts, you realize that there is a 200 GB subfolder you do not want to synchronize, uncheck it and the data is gone.

You can put sync on pause, go to file manager, move this folder somewhere else, continue to sync.

**Comment by b12f** [comment]
> You can put sync on pause, go to file manager, move this folder somewhere else, continue to sync.

That's like telling someone they should reboot their pc if nextcloud crashes instead of fixing the cause of the crash. This is at its core a UX problem, and the solution you're offering is still hugely bad UX, especially because it would still mean people lose valuable data if they think they can change syncing options in the nextcloud UI, and then don't understand the current warning (which a lot of people don't)

**Comment by sadr0b0t** [comment]
> especially because it would still mean people lose valuable data if they think they can change syncing options in the nextcloud UI, and then don't understand the current warning (which a lot of people don't)

My suggestion is to forbid removing unsynced data from nextcloud ui. The user should not be able to uncheck folder which not synced.

**Comment by wsw70** [comment]
> You can put sync on pause, go to file manager, move this folder somewhere else, continue to sync.

I install Nextcloud, start the sync, realize that what is being synced is not what I want to sync, uncheck - read the warning that states that the files in the unchecked folder will be removed and not synced anymore - but hey, they are not synced yet, because I just started the first sync after installation.

Crap, they managed to get synced finally... Good thing I have a backup /s

What I am trying to say is that once the sync is done then maybe the warning is fine (I could live with that, the idea being that I d not want **anymore** these files to be in Nextcloud) - but not at the install stage while the sync is going on.

As for the prompt itself, it should leave the choice for the files to be removed or stay as they are (not synced anymore, with some consequences down the road such as diffs between files if they are re-synced.)

**Comment by wsw70** [comment]
> My suggestion is to forbid removing unsynced data from nextcloud ui. The user should not be able to uncheck folder which not synced.

This is an interesting suggestion - fair point. I just wonder how it could work for selective sync of files within a folder

**Comment by irisv** [comment]
> You can put sync on pause, go to file manager, move this folder somewhere else, continue to sync.

You're imagining a user with perfect knowledge like yourself. But this usage case involves people who have just installed Nextcloud  and added a folder they didn't mean to add. The UI should also be safe to use for them.   

>My suggestion is to forbid removing unsynced data from nextcloud ui. The user should not be able to uncheck folder which not synced.

That would solve the issue of lost data, but if if I understand you correctly then in this scenario where a user unwittingly added a 200 GB subfolder they do not want to synchronize, that would involve uploading that 200 GB subfolder first, then the user would uncheck the box, not understanding the full implications of the warning, resulting in that 200 GB of local data being deleted. Then I imagine that user has a minor panic attack, until they realise the data is safe on the server. But then they have to download that 200 GB again and remove it from the server, just to get back to where they started. So that's 400 GB of unnecessary and unwanted traffic and one very irritated new user.

**Comment by sadr0b0t** [comment]
>I just wonder how it could work for selective sync of files within a folder

NextCloud just does have partial syncing feature as far as I know. Unchecked folder in NextCloud UI just means that this folder exists on server but the client does not want to download it as local copy. It is not about partial syncing. Current UI allows to lose data by accident and I also think this is a bug.

For having partial sync for some data in enabled folders - I think this could be discussed as a separate feature (for example, this can be implemented as a king of .gitignore file in git).

**Comment by sadr0b0t** [comment]
> then the user would uncheck the box, not understanding the full implications of the warning, resulting in that 200 GB of local data being deleted

No, the user should not be able to uncheck the box for the folder which is not synced - it should drawn in disabled state with some kind of (optional) explanation why it is disabled.

**Comment by sojer** [comment]
![image](https://user-images.githubusercontent.com/3913323/100353134-5c2b5e00-2fee-11eb-8956-a9174127b777.png)

Isn't it possible to change this dialog in such a way?
- "Disable sync but untouch locally"
  Bevor the action starts, this folder/file is automatically added to the ignored files

- "Disable sync and remove locally"
  Proceed as currently

- "Cancel"
  Just stop

**Comment by irisv** [comment]
> 
> 
> > then the user would uncheck the box, not understanding the full implications of the warning, resulting in that 200 GB of local data being deleted
> 
> No, the user should not be able to uncheck the box for the folder which is not synced - it should be drawn in disabled state with some kind of (optional) explanation why it is disabled.

But after the folder has synced, the box would be enabled again, right? You're proposing unsynced folders have the checkbox disabled? So you have to sync first, before you can remove it? That's what I meant by uploading that 200 GB first:

> in this scenario where a user unwittingly added a 200 GB subfolder they do not want to synchronize, that would involve 
uploading that 200 GB subfolder first _[ie, completing the sync, then the disabled checkbox would become active again once completed]_ then the user would uncheck the box, not understanding the full implications of the warning, resulting in that 200 GB of local data being deleted. Then I imagine that user has a minor panic attack, until they realise the data is safe on the server...

So in this usage scenario, the user is unticking the box AFTER completing the sync. Sorry if that wasn't clear.

**Comment by irisv** [comment]
> 
> 
> ![image](https://user-images.githubusercontent.com/3913323/100353134-5c2b5e00-2fee-11eb-8956-a9174127b777.png)
> 
> Isn't it possible to change this dialog in such a way?
> 
>     * "Disable sync but untouch locally"
>       Bevor the action starts, this folder/file is automatically added to the ignored files
> 
>     * "Disable sync and remove locally"
>       Proceed as currently
> 
>     * "Cancel"
>       Just stop

That's exactly what both [b12f ](https://github.com/nextcloud/desktop/issues/1238#issuecomment-714420754) and [I](https://github.com/nextcloud/desktop/issues/1238#issuecomment-733615358) have suggested above.

**Comment by sadr0b0t** [comment]
> So you have to sync first, before you can remove it? That's what I meant by uploading that 200 GB first:

You can either upload 200GB first or put upload on pause and move unsynced 200Gb folder in file manager somewhere else from sync dir location.

**Comment by robballan** [comment]
A. At some point the user has to be responsible for erroneous decisions. Being unable to stop a large sync does not lose data, it is merely highly inconvenient. Losing a local folder that is unsynced is potentionally far worse.

> On Nov 26, 2020, at 5:08 AM, WoJ <notifications@github.com> wrote:
> 
> 
> You should NOT be able to uncheck a folder for syncing UNTIL it has been synced upstream. That would solve it.
> 
> @b12f gives a very good case where this would not work: the initial setup of Nextcloud. The sync starts, you realize that there is a 200 GB subfolder you do not want to synchronize, uncheck it and the data is gone.
> 
> Not removing unsynced data is a solution that is safer.
> 
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub, or unsubscribe.
> 

_____

raa

**Comment by Forza-tng** [comment]
> ![image](https://user-images.githubusercontent.com/3913323/100353134-5c2b5e00-2fee-11eb-8956-a9174127b777.png)
> 
> Isn't it possible to change this dialog in such a way?
> 
> * "Disable sync but untouch locally"
>   Bevor the action starts, this folder/file is automatically added to the ignored files
> * "Disable sync and remove locally"
>   Proceed as currently
> * "Cancel"
>   Just stop

I think one of the main - **major** - point is that if the user had put data in the local folder to be synced (uploaded) to the server and that upload is not yet finished. It is very easy to assume that "fine, I don't need my local data as it is already on the server", when in fact it isn't true.

The disable sync dialog should show that it is currently uploading local changes (that there are un-synced local changes) and warn accordingly.

**Comment by vegatron** [comment]
WHY!!!
So I just lost my data due to this terrible implementation! My folder was unsynced and I lost my data! 
I wanted to sync it later, I was moving new folders to it, it had many many files, so I unchecked it to sync it later as not to slow down my connection, and now I can't recover my file!!

who thought this was a good idea!!
I'm really considering moving back to google drive, you just don't delete local data!

**Comment by b12f** [comment]
And this "feature" takes another victim.

> At some point the user has to be responsible for erroneous decisions.

I just don't think this is ever true. Bad design exists, and this is a perfect example of it. The huge discussions both in the owncloud repo and here report multiple failure cases _with permanent dataloss_ and it still feels to me like it is not being taken seriously. Even the "Delete this file" menu items on most OS's put the file in a recycle bin. You don't just permanently delete data after the user presses a grey button on a grey background that says "Apply" with a grey warning text next to it that "they should've just read".

Unfortunately, I've zero experience writing c++ code, otherwise I'd love to chip in and help. Not trying to be hostile, but I do want to stress the severity of this bug.

**Comment by robballan** [comment]
We are agreeing. I made that statement in the context that there is no system that can ever be perfectly designed to prevent users from making mistakes. However, there should be no system in which the design _leads_ the user to make a mistake. And this is one where that happens.

It seems that what nextcloud’s developers are assuming is that you will only uncheck the need for a local copy _after_ the data has already been synced up into the cloud server. But they don’t actually enforce this in the implementation, which catches the user by surprise. It appears that you can uncheck this requirement and have the local data erased _before_ it is copied to the cloud server, which makes no sense from anyone’s perspective.

The fix should certainly be that no local data is ever erased until it has been synced upstream first. It should be obvious that when a user deselects the need for a local copy, it’s precisely because he assumes the remote copy exists. It’s clear the developers assume this is true because they leave the checkbox available to be rechecked later, which would then require the remote data to be copied back to the local store. The error is that the sync engine fails to verify the existence of the remote copy before deleting the local copy.

—
raa

**Comment by vegatron** [comment]
It doesn't make sense for NextCloud app to delete local unsynched files. I can understand that it might deleted synched files, which could be an inconvience to get them back, but to permenantly delete local unsynched files is a disaster.

**Comment by rasos** [comment]
A client lost a local folder, which was in the sync tree and has been unchecked. The folder was not deleted immediately, but 4 months later after a total re-scan that was triggered after a server migration. So the client assumed, that the folder inside the sync tree does not get deleted and he could manually switch on and off server syncing. 

The only reason that the unattended could happen according to the discussion here would be that a folder is being moved into the sync folder and then stopped from syncing, before it has been uploaded yet. But why did it remain in the sync folder for so long "undiscovered" as a deletion candidate?

**Comment by Petermhen** [comment]
> WHY!!! So I just lost my data due to this terrible implementation! My folder was unsynced and I lost my data! I wanted to sync it later, I was moving new folders to it, it had many many files, so I unchecked it to sync it later as not to slow down my connection, and now I can't recover my file!!
> 
> who thought this was a good idea!! I'm really considering moving back to google drive, you just don't delete local data!

It is honestly unbelievable that there are people attempting to rationalise how this is anything but an absolute deal breaking flaw. 

The bare minimum expectation of any program is for it to not unexpectedly permanently delete your possibly invaluable data. A program that breaks this rule can within justification be classed as malware. The mind boggles.

**Comment by b12f** [comment]
Bump

**Comment by vegatron** [comment]
Bump

**Comment by irisv** [comment]
I've got another complaint about the client that is somewhat related, in that the client deletes files I didn't want deleted (that's why I'm posting it here rather than starting a new topic):
After using an outdated version of the Nextcloud database, the client deleted all the local files that weren't yet in that database. The behaviour that I would have wanted, and that I think most users would want, is that if the clients finds a local file that isn't yet in Nextcloud, it should add the file to the database rather than remove the file from the local client folder. Or at the very least, ask!

Here's what happened: I'm running NextcloudPi on a Raspberry Pi, with the data folder on a USB harddrive attached to the Pi. It also runs some other software, and as I was mucking around with some of that, I made a image of the SD card (which includes all of Nextcloud, except the data folder), to copy on an alternative SD card so I didn't run the risk of messing with my live SD. After a first failed attempt of messing with that other, unrelated software, I had another bright spark and wanted to try something new. So I again made a fresh copy of the SD using that backup image (now about a week old), and booted up my Raspberry Pi with it. However, I forgot I still had the Nextcloud client running on my laptop. The client quietly and diligently set to work deleting all the local files that I had created since making that backup image. This wasn't at all what I was expecting: I was expecting to have to run the occ files:scan command to force it to find the new files. Luckily, it didn't delete the files from the Nextcloud data folder (after all, it didn't know it had those), but boy, did this give me a fright! 

Now of course it was my own stupid fault for not exiting the client before starting Nextcloud with an outdated database, and in this case, no damage was done (besides raising my heartrate for a bit). But in a scenario where someone is restoring Nextcloud after some catastrophic event, when they more likely than not are restoring with an outdated database, the last thing you want the client to do, is delete local files! 

Conclusion: an "_always ask before deleting local files_" option in the client would be a great step forward, for this case as well as the problem this thread is devoted to.

**Comment by b12f** [comment]
Bumping again

**Comment by vegatron** [comment]
I'm still struggling with PTSD over the loss of my data

**Comment by dawnmy** [comment]
I am also a victim of this bug

**Comment by b12f** [comment]
Bumping

**Comment by WXpiero** [comment]
> This bug report did not receive an update in the last 4 weeks. Please take a look again and update the issue with new details, otherwise the issue will be automatically closed in 2 weeks. Thank you!

this issie is a giant joke， hope this bug will get fixed finally before we got totally wiped out by ww3.

**Comment by ainola** [comment]
I had removed one folder sync connection, which explicitly states that files will not be deleted. Right after that I had unchecked a half-synced directory inside a sync connection and my brain assumed the dialogue was saying the same thing. There's a consistency issue here.

Needless to say, I've also lost data due to that. Yes, I should have had snapshots, backups, etc. before embarking on what I was doing but... my nextcloud is what gets backed up, not the local PC.

**Comment by vegatron** [comment]
Welcome to the club! and sorry for your data loss.
I can't fathom the horrible UX choices and broken functionality that lets a backup software delete local files without clear warnings! and what makes matters worse, the developers won't even acknowledge this problem and issue a quick fix!

**Comment by irisv** [comment]
"It's not a bug, it's a _feature_"

Still needs to be fixed.

**Comment by BEEDELLROKEJULIANLOCKHART** [comment]
I've just lost a lot of data, suspiciously quickly after I installed the Desktop Sync Client. Please remediate this; to ever lose data without notification is unacceptable.

**Comment by piciuok** [comment]
bump :D

**Comment by andrejberg** [comment]
Yesterday this bug deleted work of 5 months for me. I agree with what was stated here before: a bug like that in a software that is used to !!!BACKUP!!! data is absolutely fatal and I can't believe that it is still an issue after it was reported 2019.

**Comment by BEEDELLROKEJULIANLOCKHART** [comment]
@andrejberg, the Nextcloud Desktop Sync Client does not exist to duplicate. It exists to synchronize. You should duplicate your information into somewhere that is not synchronized if you want that information to be reliably possible to recover.

**Comment by irisv** [comment]
> @andrejberg, the Nextcloud Desktop Sync Client does not exist to duplicate. It exists to synchronize. You should duplicate your information into somewhere that is not synchronized if you want that information to be reliably possible to recover.

On the first sync, synchronizating with an empty folder IS duplication of that folder.

"It's your own fault for not reading all  documentation before starting" is not a helpful comment to someone who has just lost a lot of data. Especially in a thread full of people who were confronted with the same issue.The software is counter-intuitive and should be fixed.

**Comment by ainola** [comment]
Indeed, while NC/Dropbox et al are not technically backup solutions, they provide a data-recovery-like interface in the form of the trash bin or version control.

Regardless of the "properness" of using these solutions for backups, the application should not obliterate the user files that aren't actively tracked without *any* recourse.

**Comment by BEEDELLROKEJULIANLOCKHART** [comment]
@irisv, is https://github.com/nextcloud/desktop/issues/1238#issuecomment-1172456075 response to my previous comment? I stated none of what you quote, and have stated previously at https://github.com/nextcloud/desktop/issues/1238#issuecomment-1143718052 that I have additionally lost information because this problem exists.

**Comment by irisv** [comment]
> @irisv, is "[#1238 (comment)](https://github.com/nextcloud/desktop/issues/1238#issuecomment-1172456075)" response to my previous comment? I stated none of what you quote, and have stated previously at "[#1238 (comment)](https://github.com/nextcloud/desktop/issues/1238#issuecomment-1143718052)" that I have additionally lost information because this problem exists.

I know you are also a victim. I'm sorry for your loss. I'm sorry if that wasn't clear.

The line I put in quotation marks ("It's your own fault for not reading all documentation before starting") is my (cynical) interpretation of the Nextcloud line on this issue. It's been going for years and all they can do is blame the ever growing line of victims.

Your line, "Nextcloud Desktop Sync Client does not exist to duplicate. It exists to synchronize" is similarly a distinction that most at home users won't make. I understand you're trying to explain what andrejberg _should have_ done, but its too late for that now. The point of the Nextcloud client is to synchronise your files to your server -- so you'll have a backup copy. First making a backup, as you suggest, is a very good point. But the problem is that all the people who lost data, had no idea they needed to make a backup before they started. They probably could have known, had they read all the appropriate documentation. But my point is: the client is counter-intuitive and even if you don't fully know what you're doing, using it should never EVER lead to data loss.

**Comment by WXpiero** [comment]
> I know that I made a mistake by relying completely on Nextcloud to backup such important data. However, the current clients' default behaviour to DELETE the data if not synced is not only unintuitive but also violates principles of data preservation that have to be applied to software like that. The default behaviour should be: don't touch the data. Especially if the data is newer than the one on the server. I guess this arrogant thinking of: It's your fault. You should read the documentation and backup your data properly is the reason why this bug is still unresolved and continues to cause data loss.

dude, try to use the NC sync funtion the other way around,and you will no longger facing any sync problem any more. I was using the NC sync program to delete redundant files between 2 servers for months, it works perfectly! For backup part,I use sycnthing ,which is safe and stable, and I think thats the bottom line for most of us talking about synchronize. But maybe we truly are all wrong, for some other people, the data safty is not the most prioritize factor in synchronization.

**Comment by b12f** [comment]
Bump

**Comment by sadr0b0t** [comment]
while some users post their stories how they lost their data and some others post their thoughts on how program should behave, it looks like that the problem is not that "developers" don't care about some lost data, but that they just don't know how to fix it in software core without breaking the whole thing that was forked from another project.

**Comment by vegatron** [comment]
+ bump

**Comment by vegatron** [comment]
And another bump

**Comment by s1m-e0n** [comment]
For me, the client does follow a pretty clear and practical synchronization behaviour that defines fundamentally what NC is, but this is obviously not clear to many new users, including myself when I started using NC. I'm not involved in it's development therefore I lack the necessary insight for this, but I find it really hard to imagine a different design that doesn't fundamentally change how NC works, and possibly the developers don't see a way either. Having used NC for a while now I understand and even like how it works.. but I had to learn it the hard way.
Rather than changing the behaviour, why not just display a very prominent dialogue which exactly explains how NC interprets synchronization (when the app is first launched?), possibly even - in the case where data is about to be deleted - require a box to be checked or force the user to type "yes, I understand" before any deletion happens?

**Comment by s1m-e0n** [comment]
I thought perhaps at step 4. from "Steps to reproduce" in the issue. The "yes, i understand" was just an idea, anything more prominent but especially more explanatory regarding the underlying mechanism could perhaps prevent bad things from happening.

**Comment by mingsu** [comment]
**WHERE CAN I FIND THE DELETED FOLDER??!!**

**Comment by dirdi** [comment]
> **WHERE CAN I FIND THE DELETED FOLDER??!!**
Your files and folders have been deleted permanently.

**Comment by mingsu** [comment]
> > **WHERE CAN I FIND THE DELETED FOLDER??!!**
> 
> Your files and folders have been deleted permanently.

Thanks for letting me know this.


**WHAT IS THE LOGIC OF DOING SO?! @NEXTCLOUD**

**Comment by irisv** [comment]
Depending on your operating system, there are ways to recover deleted files. Google is your friend. Do it now, because the more your computer rewrites on your harddrive, the fewer files you'll be able to recover.

And as for Nextcloud's rationale - that's what we've all been wondering about in this thread for years. Apparently it's your own fault for not reading the instructions thoroughly enough. I think the way the client behaves is counterintuitive, and your experience is another piece of evidence for that.

**Comment by mingsu** [comment]
> Depending on your operating system, there are ways to recover deleted files. Google is your friend. Do it now, because the more your computer rewrites on your harddrive, the fewer files you'll be able to recover.
> 
> And as for Nextcloud's rationale - that's what we've all been wondering about in this thread for years. Apparently it's your own fault for not reading the instructions thoroughly enough. I think the way the client behaves is counterintuitive, and your experience is another piece of evidence for that.

Thanks.

It is not hard to know that many users have been suffered this problem for long time.

The instruction is not instructive at all, what does `Cancel` mean?
- To my understanding, `Cancel` means I can't uncheck the folder that I don't want to sync. Obviously, that is not the choice for user. 
- Second, users just simply trust the APP, that of course not do suck risky deletion.

In addition, the APP doesn't have permission to delete the folder that has been update by the user after.

SO, My suggestion:

- Make the instruction more instructive
- Warn users BEFORE the APP execute the deletion
- or the best, don't delete the uncheck folders

**Comment by irisv** [comment]
>     * or the best, don't delete the unchecked folders

Indeed. If the Nextcloud client can be made to ignore certain files, surely it can ignore a specific subfolder. To many users, selecting a folder intuitively means "sync this folder" and unticking a subfolder means "ignore this folder". Unticking that subfolder doesn't intuitively correspond to "delete this folder".

And if it really can't deal with a subfolder the user doesn't want to sync, inside a folder that does need to be synced, it should offer to move that subfolder to a new destination outside the synced folder.

**Comment by Petermhen** [comment]
> > > **WHERE CAN I FIND THE DELETED FOLDER??!!**
> > 
> > 
> > Your files and folders have been deleted permanently.
> 
> Thanks for letting me know this.
> 
> **WHAT IS THE LOGIC OF DOING SO?! @nextcloud**

Apparently it's your fault for not reading the documentation before using the program (as apparently everyone does for every app).

Alsond for not understanding the precise workings of the program as well as the developers. It's your fault. Insert additional gaslighting here.

**Comment by ainola** [comment]
Hey guys, welcome to another edition of the bump channel!

**Comment by hcoin** [comment]
I agree the default sentiment is local files should not be deleted by default, and that should an 'unsynced' tree be 'resynced' nextcloud should make the user confirm all conflicts.  This is the behavior when a new user creates a folder that has some content, decides to join it to the 'synced world', and has to decide should the local content be different than the server which version to keep.

One element missing from all the upstream discussion is:  As 'cloud first' overlooks nothing originated in the cloud, the cloud is ever and always a forwarding and sync engine, that's the unshakable ground for why end users can't believe a 'delete first' policy exists in nextcloud.  

Another major thing missing from the upstream is:  Note Nextcloud in the docs explains it is not to be looked to as a backup solution.  There's a certain humility there, that is appropriate.  Now, what happens when an old database, even just a few hours or days, gets restored owing to some calamity, but the files in the directory tree have different checksums?   Databases are almost never stored in the same way as the client file tree.   A client might reasonably think when the server 'has gone away' -- 'you know, I want to decide when it comes back whether to keep my changes or to accept what's there, so I'll unsync this folder, and when I'm satisfied some new php update or random tech thing I don't understand gets fixed by techs somewhere, I'll sync it back again after it's up a good long while'.    Then as matters stand they lose their authoritative and other data, and have nextcloud admins tell them to enjoy the 'checksums are different' complaints they can do nothing about.   Which has folks looking for other solutions than own or next cloud.

"Cloud first" must appreciate the reality nothing originated there, and as nextcloud doesn't propose to be a backup solution, and as such notes it is subject to catastrophe itself, leading to such as when the database doesn't match the file tree--- deleting user files by default upon unsync is not a compatible with acceptance long term.

**Comment by ainola** [comment]
It'd be great if someone on the nextcloud team could prevent this from being closed.

**Comment by vegatron** [comment]
> It'd be great if someone on the nextcloud team could prevent this from being closed.
+1

**Comment by allexzander** [comment]
Does anyone have clear repro-steps that always lead to file deletion? I have a PR here for a similar scenario https://github.com/nextcloud/desktop/pull/5459 that might fix the current issue but would be nice to have some steps to validate if this applies to the current one.

**Comment by b12f** [comment]
1. Have a (large) existing directory with filled with directories & files
2. Connect that directory to nextcloud sync
3. Enable a large subdirectories to be synced
4. Before/during the initial sync, disable sync for one of the directories by unchecking it
5. This folder will be deleted. If it wasn't completely synced to the server, it causing the unsynced data to be completely lost.

Deletions also skip e.g. the Trashbin on Windows, so by enabling sync and toggling some settings, you're irretrievably losing data.

**Comment by allexzander** [comment]
> bumpity bump
> 
> Can we confirm #5459 solves this issue?

Yes, I can no longer reproduce.

**Comment by vegatron** [comment]
Another bumpity bump.. Such horrible design flaw must be fixed.

**Comment by vegatron** [comment]
Bump #999

**Comment by oliv3r** [comment]
Just dropping some pennies here,

I was flabbergasted that unchecking the checkbox wanted to remove local files. I could cancel the action just in time, becuase otherwise I would have lost years worth of work, just because I didn't want to sync 1 folder of a whole tree. That's just insane and horrible UX. To accomplish the same, I have to fumble with exclude patterns, and just hope I get it right. Again horrible UX.

I think what's generally missing, is a treeview, just like the select, an exclude. This would help a lot already but would lead to a confusing UI, in the main tree view, we now select something, that we then unselect in the exclude tree.

I recon the confusion comes from the fact that a checkbox is used, where something that has multiple-choices is needed. From a clicky point of view, I get the desire to have a checkbox, though not sure how easy it is (and how confusing that would be) to have a checkbox that can do multiple things. A drop-down would be far more accurate in that sense.

[sync]
[ignore] (do not sync, but do not delete)
[unsync] (delete local files)

is in short what seems to be what is missing. Ignoring a folder is what everybody here is after. If that means, internally it just adds something to the exclude list, that's an implementation detail.

I suppose the checkbox can always be combined with a checkbox, the checkbox has its own behavior, but then 'enables' the drop-down to select 'unsync/ignore'?

Having said that, having a confirmation 'going to delete <x bytes in y folders>' would still give the user a moment of 'uhm, that's not what I thought this would do'. Deletion is a destructive operation, potentially loosing years/lots of money' worth of data, Even though the sentiment is 'but the user unclicked it' shouldn't mean 'ok lets just throw it all way without confirmation'. Heck even a file browser asks confirmation.

**Comment by janvlug** [comment]
Could this important bug be ignored by the stale bot?

**Comment by rokejulianlockhart** [comment]
Better yet, get rid of it entirely.

**Comment by ainola** [comment]
Bump. Ever hear of a pangolin? They're pretty cute:

https://duckduckgo.com/?q=pangolin&t=fpas&iax=images&ia=images


Jul 16, 2023 09:34:30 github-actions[bot] ***@***.***>:

**Comment by irisv** [comment]
Is this issue really resolved? Can anyone confirm?

**Comment by rokejulianlockhart** [comment]
> This bug report is getting automatically closed due to no answer since the issue has been staled. Thank you!

That does not seem like a predefined response.

**Comment by vegatron** [comment]
Was this resolved? I stopped using it after it wiped important folders for me, and won't reinstall until its fixed

**Comment by ainola** [comment]
It wasn't. The bot closed it. I tried pinging someone on IRC but there wasn't a response, sadly. I'll try again today.

Aug 23, 2023 02:56:26 Ahmed Reeh ***@***.***>:
